### PR TITLE
Removal of pyenv

### LINKED ISSUE #32504: Remove pyenv
Rationale
* openpilot itself shouldn't care too much about which Python it runs
* since pyenv builds python from source, it adds a bunch of dependencies that get lumped in with openpilot's own dependencies
* users should choose how to manage their python env

I think this will be doable once we're on 24.04, which uses Python 3.12. AGNOS may jump to future Python versions for speedups, but openpilot should be mostly agnostic to version for core functionality.

### PR DESCRIPTION
**Pull request by BBBmau**
Resolves #32504

### CODE REVIEW

File: tools/install_ubuntu_dependencies.sh [hunk 1]
```diff
@@ -99,11 +99,12 @@ function install_ubuntu_lts_latest_requirements() {
 # Install Ubuntu 20.04 packages
 function install_ubuntu_focal_requirements() {
   install_ubuntu_common_requirements
-
+  $SUDO add-apt-repository ppa:deadsnakes/ppa
```
Comment (adeebshihadeh) [inline tools/install_ubuntu_dependencies.sh:102]: it's a bit hacky, so I want to split this out as a separate script and prompt the user to see if they want to install it. then with 24.04, we just remove this script entirely

File: tools/mac_setup.sh [hunk 2]
```diff
@@ -59,8 +59,6 @@ brew "libusb"
 brew "libtool"
 brew "llvm"
 brew "openssl@3.0"
-brew "pyenv"
-brew "pyenv-virtualenv"
```
Comment (adeebshihadeh) [inline tools/mac_setup.sh:63]: we can install "python@3.11" here

File: .python-version
Comment (adeebshihadeh) [inline .python-version:1]: @gregjhogan afaik this is a pyenv specific file, and I think it makes sense to remove it. the supported versions are specific in pyproject.toml

File: tools/install_ubuntu_dependencies.sh (hunk 1 above)
Comment (BBBmau) [reply tools/install_ubuntu_dependencies.sh]: latest commit successfully builds DockerImage with ppa 3.11, i imagine tests should pass.

Prompt is similar to `extra_packages` flow. But If the user opts out to not wanting to use deadsnakes will it just do nothing? that's what its currently doing.

File: .python-version
Comment (BBBmau) [reply .python-version:1]: yes you're right, pyenv depends on `.python-version` and can be removed entirely.

File: tools/install_ubuntu_dependencies.sh [hunk 3]
```diff
@@ -99,13 +99,36 @@ function install_ubuntu_lts_latest_requirements() {
...
+  fi
+  if [[ "$INSTALL_DEADSNAKES_PPA" == "yes" ]]; then
+    $SUDO apt-get install software-properties-common -y --no-install-recommends
+    $SUDO add-apt-repository ppa:deadsnakes/ppa
+    $SUDO apt-get install -y --no-install-recommends \
+    python3.11-dev \
+    python3.11 \
+    python3.11-venv \
+    python3.11-distutils \
+    scons \
```
Comment (adeebshihadeh) [inline tools/install_ubuntu_dependencies.sh:125]: this is the wrong scons. needs to be the one pip installs

File: tools/install_ubuntu_dependencies.sh [hunk 4]
```diff
@@ -99,13 +99,36 @@ function install_ubuntu_lts_latest_requirements() {
...
+    echo ""
+    if [[ $REPLY =~ ^[Yy]$ ]]; then
+      INSTALL_DEADSNAKES_PPA="yes"
+    fi
+  fi
+  if [[ "$INSTALL_DEADSNAKES_PPA" == "yes" ]]; then
+    $SUDO apt-get install software-properties-common -y --no-install-recommends
+    $SUDO add-apt-repository ppa:deadsnakes/ppa
+    $SUDO apt-get install -y --no-install-recommends \
+    python3.11-dev \
```
Comment (adeebshihadeh) [inline tools/install_ubuntu_dependencies.sh:121]: make sure to indent these one level

**Comment by BBBmau** [comment]
latest commit is just to confirm tests pass. I'll do one more cleanup before marking as ready

File: selfdrive/test/ci_shell.sh [hunk 5]
```diff
@@ -11,6 +11,7 @@ fi
 
 docker run \
        -it \
+       --shm-size=100m \
```
Comment (BBBmau) [inline selfdrive/test/ci_shell.sh:14]: I had to increase the memory usage due to getting a `Fatal Python error: Bus error`

https://github.com/commaai/openpilot/actions/runs/9201668216/job/25310251273

**Pull request merged**